indico/modules/events/contributions/client/js/index.jsx

```diff
@@ -29,30 +30,42 @@ import PublicationSwitch from './PublicationSwitch';
     if (!editableSubmissionButton) {
       return;
     }
-
     const availableTypes = JSON.parse(editableSubmissionButton.dataset.availableTypes);
     const {eventId, contributionId, contributionCode} = editableSubmissionButton.dataset;
 
-    let responses;
+    let responses, lastRevFiles;
     try {
-      responses = await Promise.all(
-        availableTypes.map(type => indicoAxios.get(fileTypesURL({confId: eventId, type})))
-      );
+      responses = await Promise.all([
+        Promise.all(
+          availableTypes.map(type => indicoAxios.get(fileTypesURL({confId: eventId, type})))
+        ),
+        indicoAxios
+          .get(paperInfoURL({confId: eventId, contrib_id: contributionId}))
+          // eslint-disable-next-line no-unused-vars
+          .catch(e => undefined),
+      ]);
     } catch (e) {
       handleAxiosError(e);
       return;
     }
-
     const fileTypes = _.fromPairs(
-      responses.map((response, index) => [availableTypes[index], camelizeKeys(response.data)])
+      responses[0].map((response, index) => [availableTypes[index], camelizeKeys(response.data)])
     );
 
+    if (responses[1]) {
+      const {isInFinalState, lastRevision} = camelizeKeys(responses[1].data);
+      if (isInFinalState && lastRevision) {
+        lastRevFiles = lastRevision.files;
+      }
+    }
+
     ReactDOM.render(
       <EditableSubmissionButton
         fileTypes={fileTypes}
-        eventId={eventId}
-        contributionId={contributionId}
+        eventId={Number(eventId)}
```

Comment:
`parseInt(eventId, 10)`

Reply:
Is there any difference, isn't `Number` cleaner?

Reply:
I think in this case it [doesn't really matter](https://stackoverflow.com/questions/4564158/what-is-the-difference-between-parseintstring-and-numberstring-in-javascript). You could even use the `eventId={+eventId}` shortcut ;)

When dealing with user input, `+` and `Number` have the disadvantage of converting `0x*` and (I think) `0*` (octal) as well which is confusing if you type e.g. `0123` and get `83`.

Reply:
Totally forgot about the `+`. It seems like a good idea considering ids should be safe here!